openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -187,6 +193,13 @@ def to_representation(self, user):  # lint-amnesty, pylint: disable=arguments-di
                 }
             )
 
+        if activation_key:
+            data.update(
+                {
+                    "activation_key": activation_key,
+                }
+            )
```

Comment:
As Awais said, we can set this above if `activation_key` is either `None` or a valid activation key.